### ISSUE TITLE
Optionally create tibbles instead of data frames in fromJSON()

### ISSUE THREAD

**Issue by krlmlr** [jeroen/jsonlite#356]
Converting all nested data frames to tibbles is a very cumbersome exercise, tibble's output is more compact for nested data. Would you consider an optional argument to `fromJSON()` ?

**Comment by jeroen** [comment]
Does it suffice if we add a `tibble` class to data frames, or do we really need to depend on the tibble package?

**Comment by krlmlr** [comment]
We could add a `tibble = FALSE` argument and convert to tibbles only if `tibble = TRUE` ? We can then get by with only suggesting tibble, not importing it. We'd check with `requireNamespace()`, this has the side effect of loading the package, which makes all methods available.

I'd advise against bluntly adding the class without loading tibble.

**Comment by jeroen** [comment]
What is the disadvantage of just adding the `tibble` class to the data frame? Adding tibble to suggests adds 12 dependencies to CMD check, that is quite expensive for a function that does the same thing.

**Comment by krlmlr** [comment]
``` r
data <- data.frame(a = 1)
data[, "a"]
#> [1] 1
class(data) <- c("tbl_df", "tbl", "data.frame")
data[, "a"]
#> [1] 1
requireNamespace("tibble", quietly = TRUE)
data[, "a"]
#> # A tibble: 1 x 1
#>       a
#>   <dbl>
#> 1     1
```

<sup>Created on 2021-06-09 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

Also: https://twitter.com/krlmlr/status/1067856118385381377

How about giving the user control over the data frame type, with `df_convert = identity` being the default? The user could add `df_convert = as_tibble` or any other coercing function, without adding a dependency to jsonlite.

**Comment by jeroen** [comment]
> How about giving the user control over the data frame type, with df_convert = identity being the default? The user could add df_convert = as_tibble or any other coercing function, without adding a dependency to jsonlite.

Yes that sounds good. That will also make the data.table people happy then.

**Comment by krlmlr** [comment]
Should we support `simplifyDataFrame = as_tibble` instead of adding a new argument? The current `simplifyDataFrame = TRUE` would be the same as `simplifyDataFrame = as.data.frame` . Happy to take a stab after #357.
